custom/icds_reports/utils/aggregation_helpers/distributed/agg_child_health.py

```diff
@@ -21,46 +26,53 @@ def aggregate(self, cursor):
         update_queries = self.update_queries()
         rollup_queries = [self.rollup_query(i) for i in range(4, 0, -1)]
 
-        cursor.execute(f"DROP TABLE IF EXISTS {self.staging_tablename}")
+        self.cleanup(cursor)
         # create staging table
-        cursor.execute(f"CREATE UNLOGGED TABLE {self.staging_tablename} (LIKE {self.base_tablename})")
-        # initial inserts into staging table
+        cursor.execute(f"""
+        CREATE TABLE {self.staging_tablename} (
+            LIKE {self.base_tablename}
+            INCLUDING INDEXES INCLUDING CONSTRAINTS INCLUDING DEFAULTS,
+            CHECK (month = DATE '{self.month.strftime('%Y-%m-01')}')
+        )
+        """)
+
+        logger.info('start agg_child_health data staging')
         for staging_query, params in staging_queries:
             cursor.execute(staging_query, params)
         # update staging table
         for query, params in update_queries:
             cursor.execute(query, params)
+        logger.info('start agg_child_health data rollup')
         for query in rollup_queries:
             cursor.execute(query)
+        logger.info('finish agg_child_health data rollup')
+        logger.info('finish agg_child_health data staging')
 
-        # create new monthly table if it does not exist
-        cursor.execute(f"""
-        CREATE TABLE IF NOT EXISTS "{self.monthly_tablename}" (
-            CHECK (month = DATE '{self.month.strftime('%Y-%m-01')}')
-        )
-        INHERITS ({self.base_tablename})
-        """)
-        for index_query in self.indexes():
-            cursor.execute(index_query)
-
+        logger.info('start agg_child_health table swap')
         from custom.icds_reports.models import AggChildHealth
         db_alias = db_for_read_write(AggChildHealth)
         with transaction.atomic(using=db_alias):
-            # drop old style tables if they exist
-            for i in range(1, 6):
-                cursor.execute(f'DROP TABLE IF EXISTS "{self._tablename_func(i)}"')
-            # delete data from monthly table
-            cursor.execute(f'DELETE FROM "{self.monthly_tablename}"')
-            # insert into monthly table from staging table
+            # Detach the previous table
+            cursor.execute(f'ALTER TABLE IF EXISTS "{self.monthly_tablename}" NO INHERIT "{self.base_tablename}"')
+            # Rename the previous table
             cursor.execute(f"""
-            INSERT INTO "{self.monthly_tablename}" (
-                SELECT * FROM "{self.staging_tablename}"
-                ORDER BY aggregation_level, state_id, district_id, block_id, supervisor_id, awc_id
-            )
+            ALTER TABLE IF EXISTS "{self.monthly_tablename}" RENAME TO "{self.temporary_table_to_be_deleted}"
```

Comment:
Why is it being put into this temp table?  this table is not looking used anywhere. Is it because to be used as a fallback in case of failure and prevent data loss?

Reply:
So as of right now (in my new iteration https://github.com/dimagi/commcare-hq/pull/25664) this wouldn't happen because I run `cleanup` before and after. We may definitely do something like that in the future (Clayton and I talked about this briefly today), but I think we'd want to have more tables like agg_awc and agg_ccs_record also follow a similar process before offering it up as a solution to GS. I imagine there could be a workflow where its something like:

1. bug gets deployed
2. aggregation happens
3. a table looks strange
4. revert to yesterday's table